### DISCUSSION TITLE
How to integrate fastlane with github actions to auto deploy to testflight for testing.

### DISCUSSION THREAD

**Discussion by liudonghua123** [fastlane/fastlane#16890] [Q&A]
I want to add `ci`/`cd` support for my flutter apps, I create a simple flutter repo (https://github.com/liudonghua123/flutter_app), and after searched and tried some docs/instructions on the internet, now my `github workflow` file is https://github.com/liudonghua123/flutter_app/blob/master/.github/workflows/testflight.yml, and `Fastfile` is https://github.com/liudonghua123/flutter_app/blob/master/ios/fastlane/Fastfile, and I can deploy it to testflight, but a few minutes later, I received email shown that `ITMS-90426: Invalid Swift Support - The SwiftSupport folder is missing. Rebuild your app using the current public (GM) version of Xcode and resubmit it.`. I have tried a lot, but none of theme work for me.

![image](https://user-images.githubusercontent.com/2276718/87865847-959eed80-c9ac-11ea-857a-e0b652b41ba8.png)

**Comment by joshdholtz** [comment]
@liudonghua123 Hey 👋 It looks like this may not be an issue with _fastlane_ specifically but possibly something in the flutter or Xcode project 🤔 Are you able to successfully submit this to TestFlight when building and uploading manually?

**Comment by aprilmintacpineda** [comment]
I am also looking to do the same thing, I can deploy on my mac by running `bundle exec fastlane dev` which will run the `dev lane`. The tricky part is that when running that on CI/CD environment, you actually need to login and you need to put in the right certs and stuff... which I haven't been able to do

**Reply by joshdholtz** [reply]
For certs and profiles you should look into using _match_ 😊 - https://docs.fastlane.tools/actions/match/#match

For authorizing requests and uploading you should switch over from using Apple ID to the App Store Connect API Key 🙃 - https://docs.fastlane.tools/app-store-connect-api/

I have a blog post and a _fastlane_ plugin for setting up _match_ inside of GitHub Actions 😉  - https://www.joshholtz.com/blog/2020/05/19/fastlane-plugin-for-setting-up-github-actions.html

Hopefully this is enough information to get you started!

**Reply by aprilmintacpineda** [reply]
I will take a look at this 🙇🏻  Thank you very much, you are awesome! 👏
